src/sentry/utils/tag_normalization.py

```diff
@@ -0,0 +1,92 @@
+import functools
+import re
+
+_KNOWN_TAGS = {
+    "sentry.cocoa",
+    "sentry.dart",
+    "sentry.dotnet",
+    "sentry.elixir",
+    "sentry.go",
+    "sentry.java",
+    "sentry.javascript.angular",
+    "sentry.javascript.browser",
+    "sentry.javascript.capacitor",
+    "sentry.javascript.cordova",
+    "sentry.javascript.deno",
+    "sentry.javascript.electron",
+    "sentry.javascript.ember",
+    "sentry.javascript.gatsby",
+    "sentry.javascript.miniapp",
+    "sentry.javascript.nextjs",
+    "sentry.javascript.node",
+    "sentry.javascript.node.experimental",
+    "sentry.javascript.node.hapi",
+    "sentry.javascript.raycast",
+    "sentry.javascript.react",
+    "sentry.javascript.react.native",
+    "sentry.javascript.remix",
+    "sentry.javascript.serverless",
+    "sentry.javascript.sfcc",
+    "sentry.javascript.svelte",
+    "sentry.javascript.sveltekit",
+    "sentry.javascript.vue",
+    "sentry.kubernetes",
+    "sentry.lua",
+    "sentry.native.android",
+    "sentry.native.dotnet",
+    "sentry.native.unity",
+    "sentry.native.unreal",
+    "sentry.objc",
+    "sentry.perl",
+    "sentry.php",
+    "sentry.python",
+    "sentry.ruby",
+    "sentry.rust",
+    "sentry.swift",
+}
+
+
+_SYNONYMOUS_TAGS = {
+    "sentry.android": "sentry.native.android",
+    "sentry.cordova": "sentery.javascript.cordova",
+    "sentry.electron": "sentry.javascript.electron",
+    "sentry.javascript.angular.ivy": "sentry.javascript.angular",
+    "sentry.javascript.react.expo": "sentry.javascript.react",
+    "sentry.javascript.react.native.expo": "sentry.javascript.react.native",
+    "sentry.laravel": "sentry.php.laravel",
+    "sentry.react": "sentry.javascript.react",
+    "sentry.symfony": "sentry.php.symfony",
+    "sentry.unity": "sentry.native.unity",
+    "sentrydotnet": "sentry.dotnet",
+}
+
+
+@functools.lru_cache(maxsize=300)
+def normalize_sdk_tag(tag: str) -> str:
+    """normalizing tags coming from SDKs to more managable canonical form"""
+
+    # ignore non-sentry SDK tags
+    if not tag.startswith("sentry"):
+        return "other"
+
+    # replace non-word characters with dots (normalize sentry-foo to sentry.foo)
+    tag = re.sub(r"[\W_]+", ".", tag)
+
+    # collapse known synonymous tags
+    tag = _SYNONYMOUS_TAGS.get(tag, tag)
+
+    if not tag.startswith("sentry."):
+        return "other"
+
+    # collapse tags other than JavaScript / Native to their top-level SDK
+
+    if not tag.split(".")[1] in {"javascript", "native"}:
+        tag = ".".join(tag.split(".", 2)[0:2])
+
+    if tag.split(".")[1] == "native":
+        tag = ".".join(tag.split(".", 3)[0:3])
+
+    if tag not in _KNOWN_TAGS:
```

Comment:
I think there are a few other values in `KNOWN_TAGS` which we can collapse into this `other` category. I'll do a little poking around.

Reply:
Okay, ones which turn out to be community:

`sentry.javascript.miniapp` https://github.com/lizhiyao/sentry-miniapp
`sentry.javascript.raycast` Can't find a repo for this, but the string appears nowhere in the entire `getsentry` org, so I think we're safe to collapse it
`sentry.javascript.sfcc` Same as with Raycast
`sentry.lua` Listed in our docs as third-party https://docs.sentry.io/platforms/#community-supported
`sentry.node.hapi` https://github.com/hydra-newmedia/hapi-sentry

Reply:
Ah, you're theorizing why there might not be a public repo setting those strings anywhere easily findable? Yeah, that makes sense. Anyway, I definitely think we can throw those into `other`.